spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/health/PubSubHealthIndicatorAutoConfiguration.java

```diff
@@ -34,21 +38,23 @@
  * {@link PubSubHealthIndicator}.
  *
  * @author Vinicius Carvalho
+ * @author Elena Felder
  *
  * @since 1.2.2
  */
-@Configuration
+@Configuration(proxyBeanMethods = false)
```

Comment:
What does this change do?

Reply:
Unrelated to the PR, but Spring Boot classes turn off bean proxying wherever possible for performance and to avoid tripping graal incompatibilities, so I tacked it on here.
`proxyBeanMethods` being on is only useful if you call bean-producing methods directly in autoconfiguration, which we definitely don't here.

Reply:
Filed #2500 to track.